apps/cowswap-frontend/src/modules/wallet/containers/ConnectWalletOptions.tsx

```diff
@@ -57,19 +59,13 @@ interface InjectedOptionsProps {
 }
 
 function InjectedOptions({ connectionProps, multiInjectedProviders }: InjectedOptionsProps) {
-  const isInjected = getIsInjected()
-
-  if (!isInjected) {
-    if (!isMobile) {
-      return <InstallMetaMaskOption />
-    } else {
-      return <OpenMetaMaskMobileOption />
-    }
-  } else {
-    if (multiInjectedProviders.length) {
-      return (
-        <>
-          {multiInjectedProviders.map((providerInfo) => {
+  if (multiInjectedProviders.length) {
+    return (
+      <>
+        {multiInjectedProviders
+          // Even if we detect the MetaMask Extension, we prefere to use the MetaMask SDK
+          .filter((providerInfo) => !providerInfo.info.rdns.startsWith('io.metamask'))
```

Comment:
Not 100% sure, but I _think_ this is somewhat related to this:

![image](https://github.com/user-attachments/assets/80bd426b-9747-4238-8316-3aa2bd719c08)

Injected wallets can now get duplicated.
Not reproducible on prod.

Might be related to injected wallets behaving like Metamask.

Reply:
Fixed. It was related to [this line](https://github.com/cowprotocol/cowswap/pull/5028/files#diff-aebb3e19e3823d5ba6beab9c5724f0ca47417367148900bd3bd0f80e05d971dfR29)